MonoGame.Framework/Graphics/GraphicsDevice.cs

```diff
@@ -509,6 +514,41 @@ public DepthStencilState DepthStencilState
 
                 _depthStencilStateDirty = true;
             }
+        }
+
+        /// <summary>
+        /// Get an effect and store it in the cache if necessary.
+        /// </summary>
+        /// <param name="graphicsDevice"></param>
+        /// <param name="effectCode"></param>
+        /// <param name="index"></param>
+        /// <param name="count"></param>
+        /// <param name="effectKey"></param>
+        /// <param name="headerSize"></param>
+        /// <returns></returns>
+        internal Effect GetEffect(GraphicsDevice graphicsDevice, byte[] effectCode, int index, int count, int effectKey, int headerSize)
```

Comment:
passing graphicsDevice is no longer needed. All references to graphicsDevice can be replaced with 'this.'